pkg/oauth/github/github.go

```diff
@@ -138,16 +145,20 @@ func (c *OAuthClient) decideRole(user string, teams []*github.Team) (role model.
 		}
 
 		t := fmt.Sprintf("%s/%s", org, slug)
+		if v, ok := roles[t]; ok {
+			role.ProjectRbacRoles = append(role.ProjectRbacRoles, v)
+		}
+
 		switch t {
 		case c.adminTeam:
-			role = model.Role_ADMIN
+			role.ProjectRole = model.Role_ADMIN
 			return
 		case c.editorTeam:
-			role = model.Role_EDITOR
+			role.ProjectRole = model.Role_EDITOR
 			found = true
 		case c.viewerTeam:
-			if role != model.Role_EDITOR {
-				role = model.Role_VIEWER
+			if role.ProjectRole != model.Role_EDITOR {
+				role.ProjectRole = model.Role_VIEWER
 				found = true
 			}
 		}
```

Comment:
Why do we still need to check this?
As far as I see those built-in roles are included in the return value of `c.project.GetAllUserGroups()` and then they are already being checked at line 148. 🤔

Reply:
This is to avoid breaking changes, I mean, I did like that because I thought it's better to hold old RBAC role for a while.

We're just discussing this as following.
https://github.com/pipe-cd/pipecd/pull/3666#discussion_r945600460